test/e2e/egressqos.go

```diff
@@ -229,6 +229,64 @@ spec:
 		},
 		ginkgotable.Entry("ipv4 pod", tcpdumpIPv4, &dstPod1IPv4, "/32", &dstPod2IPv4, "/32"),
 		ginkgotable.Entry("ipv6 pod", tcpdumpIPv6, &dstPod1IPv6, "/128", &dstPod2IPv6, "/128"))
+
+	ginkgo.It("Should deny resources with bad values", func() {
```

Comment:
That's actually pretty similar to how I originally wrote these tests, but switched to using files. I don't know the reason for why we don't want to import the way you suggested / or not having the tests under their own module but there probably is?

Reply:
There's also the dynamic go-client that doesn't pull in all of those dependencies: https://caiorcferreira.github.io/post/the-kubernetes-dynamic-client/
With that said, the file approach is less cumbersome and it's also what we have right now, so it wouldn't be worth it rewriting everything.

If the list of types for ovn-kubernetes grows even further, I'd be +1 for moving them to their own repository and then pulling them in as a dep for ovn-k and for any tests that we would write. 

For the time being, /lgtm

Reply:
So the reason why k8s packs its e2e like that without any dependency is so as to be able to use that binary freely, example we run k8s e2e test binary in our shard conformance tests. 
The test suites generally have their own test framework and we must be able to build an independent binary out of it and not be vendoring things in for the tests.

Reply:
Sure Andreas, I know that but these are all testframework. its inside the "test/e2e/" parts. does it actually vendor in the actually code, doing a vendor of this : github.com/ovn-org/ovn-kubernetes/go-controller doesn't seem right.

Reply:
If I read the code right, they don't have to vendor in k8s because their e2e framework is in the same module as their other production code. 
I'm making a moot point btw, I'm fine leaving it as it is. It was nice doing the exercise once though and see what it could look like.
Thanks for your input!